roles/korp-backend/defaults/main.yml

```
@@ -29,7 +29,8 @@ mk_directories:
 - { path: /data1/korp/           , owner: root, group: root, mode: "u+rwx,g+rwxs,o+rx" }
 - { path: /data1/korp/log/       , owner: root, group: root, mode: "u+rwx,g+rwxs,o+rwx" }
 - { path: /data1/korp/log/korp-py, owner: gunicorn, group: root, mode: "u+rwx,g+rwxs,o+rwx" }
-- { path: "{{ backend_cache_dir }}", owner: gunicorn, group: clarin, mode: "u+rwx,g+rwxs,o-rwx" }
+- { path: "{{ backend_cache_dir }}", owner: gunicorn, group: apache, mode: "u+rwx,g+rwxs,o-rwx" }
 - { path: "{{ korp_git_root }}", owner: root, group: clarin, mode: "u+rwx,g+rwxs,o+rx" }
 - { path: /v/appl/utils/     , owner: root, group: clarin, mode: "u+rwx,g+rwxs,o-rwx" }
-- { path: /var/www/html/korp/cgi-bin/korp/  , owner: root, group: root, mode: "u+rwx,g+rxs,o-rwx" } # for old backend korp.cgi
+- { path: /var/www/html/korp/cgi-bin/korp/  , owner: root, group: root, mode: "u+rwx,g+rxs,o+rx" } # for old backend korp.cgi
```

```suggestion
- { path: /var/www/html/korp/cgi-bin/korp/  , owner: root, group: apache, mode: "u+rwx,g+rx,o-rwx" } # for old backend korp.cgi
```
Would that be a bit more secure? The benefit would be that intruders coming other ways than via apache could not read this code. But since apache is the most likely entry point, my suggestion might be a bit academic. I'd be happy to approve the present status.

---

Myeah, I guess I wrote it like that as a defensive "now we won't be able to mess up the permissions / users / groups so that korp_download won't be able to run suddenly" thing. I could change it. Byt why the +s was ever there, I don't really know (was that how it was able to write to the cache directory in the past)?

---

> why the +s
I must plead ignorance. Maybe a copy/paste thing.

---

We can do it this way, I just better deploy it one more time then to test.